src/main/java/seedu/allonus/expense/ExpenseTracker.java

```diff
@@ -96,8 +115,81 @@ private static void addExpense(ArrayList<Expense> list, Expense e) {
+     * @param index the index of the item to be edited
+     * @param ui    ui object to collect user's inputs
+     * @throws IndexOutOfBoundsException if the expense record is not found
+     */
+    private static void editExpense(ArrayList<Expense> list, int index, TextUi ui) throws IndexOutOfBoundsException {
+        Expense toBeEdited = list.get(index - 1);
+        System.out.println(CHOSEN_EXPENSE_TO_EDIT + toBeEdited);
+        System.out.println(CHOSEN_FIELD_TO_EDIT);
+        String fieldToEdit;
+        boolean isFinishedEditing = false;
+        while (!isFinishedEditing) {
+            fieldToEdit = ui.getUserInput();
+            if (fieldToEdit.trim().equalsIgnoreCase("done")) {
+                isFinishedEditing = true;
+                System.out.println(EDITING_COMPLETE);
+                System.out.println(NEWLY_EDITED_EXPENSE_RECORD + toBeEdited);
+            } else {
+                try {
+                    editField(fieldToEdit, toBeEdited);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println(INCORRECT_NUMBER_OF_FIELDS);
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Edits a specific field of a chosen expense record based on user's input.
+     *
+     * @param fieldToEdit the field chosen by the user to edit
+     * @param toBeEdited  the expense record object to be edited
+     * @throws IndexOutOfBoundsException if new value is missing (without spaces)
+     */
+    private static void editField(String fieldToEdit, Expense toBeEdited) throws IndexOutOfBoundsException {
+        String[] newFields = fieldToEdit.split(" ");
+        String field = newFields[0];
+        String newValue = newFields[1].trim();
+        if (newValue.length() == 0) {
+            System.out.println(MSG_NEW_VALUE_CANNOT_BE_EMPTY);
+            return;
+        } else {
+            switch (field) {
+            case ("date"):
+                toBeEdited.setDate(newFields[1]);
+                System.out.println(NEW_DATE_VALUE_SET);
+                break;
+            case ("amount"):
+                toBeEdited.setAmount(newFields[1]);
+                System.out.println(NEW_AMOUNT_VALUE_SET);
+                break;
+            case ("category"):
+                toBeEdited.setCategory(newFields[1]);
+                System.out.println(NEW_CATEGORY_VALUE_SET);
+                break;
+            case ("remarks"):
+                toBeEdited.setRemark(newFields[1]);
+                System.out.println(NEW_REMARKS_VALUE_SET);
+                break;
+            case ("done"):
+                return;
+            default:
+                System.out.println(MSG_INVALID_COMMANDS);
+                return;
+            }
+        }
+    }
+
     /**
      * Determines which command to execute depending on the keyword supplied.
+     *
      * @param ui the user input itself
      * @throws ExpenseException if an invalid keyword is supplied
      */
```

Comment:
This method is more than 30 lines long, is it possible to extract some methods?

Reply:
Thanks! Refactored some methods out so functional code should be less than 30 lines now.

Reply:
Thanks, looks good!